text/0014-ai-prompt-api.md

```diff
@@ -0,0 +1,109 @@
+# RFC Template
+
+- Start Date: 2025-04-09
+- RFC PR: [electron/rfcs#14](https://github.com/electron/rfcs/pull/14)
+- Electron Issues:
+- Reference Implementation:
+- Status: **Proposed**
+
+# Prompt API support
+
+## Summary
+
+The Prompt API is part of [Built-in AI](https://developer.chrome.com/docs/ai/built-in)
+capabilities in Chromium based browsers. With the Prompt API developers can send
+natural language requests from the renderer for local processing.  This proposal adds
+support for this API in Electron.
+
+## Motivation
+
+Built in AI features are being implemented in Chromium based browsers.  Electron should
+also support these features.
+
+## Guide-level explanation
+
+The Prompt API can be used by Electron app developers by defining a Prompt API handler
+via `Session.setPromptAPIHandler`.  This handler will allow app developers to proxy
+Prompt API calls to a local LLM implementation of their choice.
+
+`ses.setPromptAPIHandler(handler)`
+* `handler` Function\<[LanguageModel](https://github.com/webmachinelearning/prompt-api?tab=readme-ov-file#full-api-surface-in-web-idl)\> | null
+  
+This handler will be called when web content calls the [Prompt API](https://github.com/webmachinelearning/prompt-api).
+
+The handler should return a [Language Model Object](https://github.com/webmachinelearning/prompt-api?tab=readme-ov-file#full-api-surface-in-web-idl).
+
+Given the demands of running LLM locally, a utility process should be used to offload processing.  For example:
+
+```js
+const { session, utilityProcess } = require('electron/main');
+
+class ElectronAI {
+  #aiProcess; //utilityProcess to handle LLM processing.  
+
+  async createInternal(options) {
+    this.#aiProcess = utilityProcess.fork(utilityScriptPath, [], {
+        stdio: ['ignore', 'pipe', 'pipe', 'pipe'],
+    });
+  }
+
+  async prompt(input = '', options) {
+    if (!this.#aiProcess) {
+      throw new Error('AI model process not started');
+    }
+    this.#aiProcess.postMessage({
+      type: UTILITY_MESSAGE_TYPES.SEND_PROMPT,
+      data: { input, stream: false, options },
+    });    
+  }
+
+  async destroy() {
+    if (this.#aiProcess) {
+      this.#aiProcess.postMessage({ type: UTILITY_MESSAGE_TYPES.STOP });    
+    }
+    this.#aiProcess = null;
+  }
+
+  async promptStreaming(input = '', options) {
+    //Use aiProcess to handle prompt streaming
+  }
+
+  static async create(options) {
+    const electronAI = new ElectronAI();
+    electronAI.createInternal(options);
+    return electronAI;
+  }
+}
+
+session.defaultSession.setPromptAPIHandler(() => { ElectronAI });
```

Comment:
If we directly pass a construction function, I think it'll make it easier to more flexibly swap out the implementation, and will also make it easier to adapt to changes in the upstream API

```suggestion
session.defaultSession.setPromptAPIHandler((webContents, options) => new ElectronAI(webContents, options));
```